bitcoin/sign.go

```diff
@@ -129,6 +129,85 @@ func (w *BitcoinWallet) buildTx(amount int64, addr btc.Address, feeLevel wi.FeeL
 	return authoredTx.Tx, nil
 }
 
+func (w *BitcoinWallet) buildSpendAllTx(addr btc.Address, feeLevel wi.FeeLevel) (*wire.MsgTx, error) {
+	tx := wire.NewMsgTx(1)
+
+	height, _ := w.ws.ChainTip()
+	utxos, err := w.db.Utxos().GetAll()
+	if err != nil {
+		return nil, err
+	}
+	coinMap := util.GatherCoins(height, utxos, w.ScriptToAddress, w.km.GetKeyForScript)
+
+	totalIn := int64(0)
+	additionalPrevScripts := make(map[wire.OutPoint][]byte)
+	additionalKeysByAddress := make(map[string]*btc.WIF)
+
+	for coin, key := range coinMap {
+		outpoint := wire.NewOutPoint(coin.Hash(), coin.Index())
+		in := wire.NewTxIn(outpoint, nil, nil)
+		additionalPrevScripts[*outpoint] = coin.PkScript()
+		tx.TxIn = append(tx.TxIn, in)
+		totalIn += int64(coin.Value().ToUnit(btc.AmountSatoshi))
+
+		addr, err := key.Address(w.params)
+		if err != nil {
+			continue
+		}
+		privKey, err := key.ECPrivKey()
+		if err != nil {
+			continue
+		}
+		wif, _ := btc.NewWIF(privKey, w.params, true)
+		additionalKeysByAddress[addr.EncodeAddress()] = wif
+	}
+
+	// outputs
+	script, err := txscript.PayToAddrScript(addr)
+	if err != nil {
+		return nil, err
+	}
+
+	// Get the fee
+	feePerByte := int64(w.GetFeePerByte(feeLevel))
+	estimatedSize := EstimateSerializeSize(1, []*wire.TxOut{wire.NewTxOut(0, script)}, false, P2PKH)
+	fee := int64(estimatedSize) * feePerByte
+
+	// Check for dust output
+	if txrules.IsDustAmount(btc.Amount(totalIn - fee), len(script), txrules.DefaultRelayFeePerKb) {
+		return nil, wi.ErrorDustAmount
+	}
+
+	// Build the output
+	out := wire.NewTxOut(totalIn - fee, script)
+	tx.TxOut = append(tx.TxOut, out)
+
+	// BIP 69 sorting
+	txsort.InPlaceSort(tx)
+
+	// Sign
+	getKey := txscript.KeyClosure(func(addr btc.Address) (*btcec.PrivateKey, bool, error) {
+		addrStr := addr.EncodeAddress()
+		wif := additionalKeysByAddress[addrStr]
+		return wif.PrivKey, wif.CompressPubKey, nil
+	})
+	getScript := txscript.ScriptClosure(func(
+		addr btc.Address) ([]byte, error) {
+		return []byte{}, nil
+	})
+	for i, txIn := range tx.TxIn {
+		prevOutScript := additionalPrevScripts[txIn.PreviousOutPoint]
```

Comment:
Handle case where value is not present?

Reply:
This one will just  error later if the script is nil.